src/sentry/migrations/0912_split_discover_dataset_dashboards_self_hosted.py

```diff
@@ -0,0 +1,83 @@
+# Generated by Django 5.2.1 on 2025-05-20 17:45
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Q
+
+from sentry.discover.dashboard_widget_split import _get_and_save_split_decision_for_dashboard_widget
+from sentry.models.dashboard_widget import DashboardWidgetTypes, DatasetSourcesTypes
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def split_discover_dataset_dashboards_self_hosted(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    DashboardWidgetQuery = apps.get_model("sentry", "DashboardWidgetQuery")
+    catch_all_unsplit_widgets = Q(
+        widget__widget_type=DashboardWidgetTypes.DISCOVER,
```

Comment:
Let's copy over all the enums into the migration file- it's a good practice to not reference application code in the migration. In case someone changes the enums in the application code, we know it's not going to break the migration if it's copied over here